modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/RuleBuilder/RulesSupport.es.js

```diff
@@ -206,6 +226,29 @@ const getExpressionFields = (action, regex = regexFieldPattern) => {
 	return action.expression.match(regex);
 };
 
+const getFieldOptions = (fieldName, pages) => {
+	let options = [];
+	const visitor = new PagesVisitor(pages);
+
+	const field = visitor.findField((field) => {
+		return field.fieldName === fieldName;
+	});
+
+	options = field ? field.options : [];
+
+	return options;
+};
+
+const getFieldType = (fieldName, pages) => {
+	return getFieldProperty(pages, fieldName, 'type');
+};
```

Comment:
@alinedoleron, a very [similar method](https://github.com/natocesarrego/liferay-portal/blob/65c170e3780026a0d25b48b4f8a697af5bccade4/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/resources/META-INF/resources/js/components/RuleEditor/RuleEditor.es.js#L598-L602) is available on `RuleEditor`, should we reuse it?

Please, analyze if makes sense reuse it here, as you did with the method `getFieldOptions`. Think if makes more sense the method be on `RuleEditor` or on `RuleSupport`. What is the purpose of these the classes? Consider it on your decision.

Reply:
I think it's better to keep it in RuleSupport, since it's also being used on RuleEditor and RuleList.